Psalm integration

jslmorrison opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description
Psalm static analysis integration

jslmorrison committed:
Add psalm xml config file

Signed-off-by: John Morrison <jslmorrison@gmail.com>

jslmorrison committed:
Update composer dev requirements

Signed-off-by: John Morrison <jslmorrison@gmail.com>

jslmorrison committed:
Update xml config with errorBaseline value

Also adds the generated psalm-baseline.xml file.

Signed-off-by: John Morrison <jslmorrison@gmail.com>

jslmorrison committed:
Add static-analysis to composer scripts

Signed-off-by: John Morrison <jslmorrison@gmail.com>

froschdesign commented:
@jslmorrison 
At the moment the Pslam baseline file contains many errors, do you see a chance to fix some of them?

Thank you for your help here and with other packages! 👍

boesing committed:
qa: finishing psalm configuration

- changes `totallyTyped=true` to `errorLevel=1` which is equivalent
- adds `psalm.xml` to `.gitignore` while renaming it to `psalm.xml.dist`
- adds `psalm.xml.dist` to `.gitattributes` so we do not ship this component with its psalm config
- remove `.coveralls.yml` as we do not provide anything to coveralls with the new CI pipeline

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing commented:
Thanks, @jslmorrison !
